### ISSUE TITLE
Service Fabric Reverse Proxy Timeout on PATCH

### ISSUE THREAD

**Issue by neraath** [Azure/service-fabric-issues#45]
I've got a service fabric stateless WebAPI service hosted through OWIN listener that supports the HTTP Verb PATCH. When I connect directly to the service (hosted in my local dev environment or in an actual Service Fabric cluster), the service behaves as expected:

```
PATCH http://localhost/api/app/trialbalances/2d9c6743-e7b1-4392-8f6b-0a46ca6d99b7/balances/80b1ba5e-5e5f-4196-8d6a-54547c68caf0

  {
    "id": "80b1ba5e-5e5f-4196-8d6a-54547c68caf0",
    "trialBalanceId": "2d9c6743-e7b1-4392-8f6b-0a46ca6d99b7",
    "group": {
        "id": "E4114B2D-3CAA-4F76-9B67-3E7F94513D35",
        "name": "Test Group",
        "number": "1400",
        "type": {
            "name": "Some Type",
            "order": 1000
        },
        "classfication": "dunno"
    }
  }

HTTP 200 OK

{
  "id": "80b1ba5e-5e5f-4196-8d6a-54547c68caf0",
  "firmId": "fa414f99-6636-4841-a9bc-8b05b38fab05",
  "clientId": 123478,
  "engagementId": "83eb1c81-d476-46af-b495-ba34d82931c4",
  "trialBalanceId": "2d9c6743-e7b1-4392-8f6b-0a46ca6d99b7",
  "account": {
    "id": "e3ad9eb0-5cc2-4022-91d6-b2ef5ce90b0d",
    "name": "Inventory",
    "number": "1200",
    "group": {
      "id": "e4114b2d-3caa-4f76-9b67-3e7f94513d35",
      "name": "Test Group",
      "number": "1400",
      "type": {
        "name": "Some Type",
        "order": 1000
      }
    }
  },
  "unadjusted": {
    "value": 954.13,
    "currency": "USD"
  }
}
```

Unfortunately, when I submit the same PATCH request through the reverse proxy (changing the URL based on the naming conventions for accessing services via the reverse proxy), it hangs. I've confirmed I can access other actions (GET, PUT, POST) on the same controller and service through the reverse proxy without fail. It seems as though PATCH just isn't behaving correctly:

```
PATCH http://localhost:19081/Audit.Api.TrialBalance/Audit.Api.TrialBalance.Service/2d9c6743-e7b1-4392-8f6b-0a46ca6d99b7/balances/80b1ba5e-5e5f-4196-8d6a-54547c68caf0

  {
    "id": "80b1ba5e-5e5f-4196-8d6a-54547c68caf0",
    "trialBalanceId": "2d9c6743-e7b1-4392-8f6b-0a46ca6d99b7",
    "group": {
        "id": "E4114B2D-3CAA-4F76-9B67-3E7F94513D35",
        "name": "Test Group",
        "number": "1400",
        "type": {
            "name": "Some Type",
            "order": 1000
        },
        "classfication": "dunno"
    }
  }
```

I can't figure out how to monitor the network traffic between the FabricApplicationGateway.exe, nor the FabricGateway.exe. I've tried to set the network proxy via the machine.config to no avail, so I'm flying a bit blind. 

Any ideas why this isn't working?

**Comment by neraath** [comment]
Any suggestions on this? We're dead in the water on this one.

**Comment by BharatNarasimman** [comment]
Hi Neraath, Sorry for the delay. This is a bug in the reverse proxy implementation. We will fix this in the next release.

**Comment by neraath** [comment]
@BharatNarasimman any update on when this will be implemented?

**Comment by neraath** [comment]
@BharatNarasimman or anyone on the team: I'm looking for an update on when this will be implemented.

**Comment by RyanElfman** [comment]
Any updates on this? I'm trying to use Docker Windows Containers with SF and I get a timeout when I navigate to the reverse proxy URL as well.

**Comment by BharatNarasimman** [comment]
@neraath : The support for PATCH is being tracked for our release coming in February.
@RyanElfman : What you are seeing might not be related to this. Are you trying to address the reverseproxy by specifying the ip address of the container or the host?

**Comment by RyanElfman** [comment]
@BharatNarasimman I have a VM running Windows Server 2016 with Windows Containers enabled. I created a sample MVC site, imaged that with Docker and have that Docker image running within SF successfully. I can get at the node specifically by going to http://localhost:30001 when inside or outside the VM (outside meaning on my VM host) but when I try going to http://<ip of vm>:19081 it doesn't work and I get a timeout error returned.

**Comment by eelkram** [comment]
@BharatNarasimman - Great to hear a fix is coming for this.  We are also experiencing this with a new set of services we are building on Service Fabric.  As a workaround we have switched the PATCH operations to PUT, but would like to fix them before we finish development.  The 5.4.164 version was released on Feb 3rd and doesn't contain the fix, can you confirm which release this will be in and when it will come out?

Thanks,
Mark

**Comment by BharatNarasimman** [comment]
@eelkram - The fix should be in 5.5* release which we are targeting for second week of march.

**Comment by eelkram** [comment]
Great, thanks for the update @BharatNarasimman .

**Comment by kavyako** [comment]
This fix is in version 5.5.216.0 and it has been rolled out. 

Thanks
Kavya.

**Comment by eelkram** [comment]
Excellent, thanks Kavya.
